lib/autoproj/ops/main_config_switcher.rb

```diff
@@ -16,16 +16,21 @@ def initialize(ws)
             EXPECTED_ROOT_ENTRIES = [".", "..", "autoproj_bootstrap",
                                      ".autoproj", "bootstrap.sh", ENV_FILENAME].to_set
 
+
+            def interactive_flag_from_env
+                ENV['AUTOPROJ_BOOTSTRAP_IGNORE_NONEMPTY_DIR'] != '1'
+            end
+
             # Verifies that {#root_dir} contains only expected entries, to make
             # sure that the user bootstraps into a new directory
             #
             # If the environment variable AUTOPROJ_BOOTSTRAP_IGNORE_NONEMPTY_DIR
             # is set to 1, the check is skipped
-            def check_root_dir_empty
-                return if ENV['AUTOPROJ_BOOTSTRAP_IGNORE_NONEMPTY_DIR'] == '1'
+            def check_root_dir_empty(interactive: interactive_flag_from_env)
+                return unless interactive
```

Comment:
linked to comment above: why not check 'ignore_nonempty_dir' or 'noninteractive' ?

Reply:
I don't understand this comment ... the `interactive_flag_from_env` method is here to abstract exactly that. It's named `interactive` as that's the name of the flag.

Reply:
If you deprecate the AUTOPROJ_BOOTSTRAP_IGNORE_NONEMPTY_DIR and also check for AUTOPROJ_NONINTERACTIVE then fine.
I mean that you can also set the env var to ignore non empty dir, but that does not necessarily mean you bootstrap noninteractive.